pkg/reconciler/inmemorychannel/dispatcher/controller.go

```diff
@@ -144,6 +178,41 @@ func NewController(
 	return impl
 }
 
+// IsReady implements ReadinessChecker.
+// It checks whether the dispatcher has registered a handler for all observed in-memory Channels.
+func (c *DispatcherReadyChecker) IsReady() (bool, error) {
+	c.Lock()
+	defer c.Unlock()
+
+	// readiness already observed at an earlier point in time, short-circuit the check
+	if c.isReady {
+		return true, nil
+	}
+
+	channels, err := c.chLister.List(labels.Everything())
+	if err != nil {
+		return false, fmt.Errorf("listing cached InMemoryChannels: %w", err)
+	}
+
+	c.isReady = c.chMsgHandler.CountChannelHandlers() >= len(channels)
```

Comment:
@antoineco if we have at least one IMC that is not `Ready`, this condition will never be true because the dispatcher does not create handlers for such channels:

https://github.com/knative/eventing/blob/9292d66e84fc5887b5ffdcb97b84321fc58c2268/pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go#L77-L80

Reply:
Nice catch, didn't think about that.